### ISSUE TITLE
Differentiability of the distribution objects.

### ISSUE THREAD

**Issue by sobalgi** [nicola-decao/s-vae-pytorch#9]
Hello,

The torch.distributions are not differentionable and hence the gradients do not propagate to encoder because of reparametrization using torch.distributions.

Hence the model is not training as expected.

I even checked the backpropagation and the KL loss doesnot seem to have gradients.

Can you please confirm this? Is the model training?

Best regards,
SB

**Comment by nicola-decao** [comment]
Hi,
I think you are worng. Some `torch.distributions` expose reparameterizable samples through the `rsample` which is differentiable (see here https://pytorch.org/docs/stable/distributions.html#pathwise-derivative).

What did you check?

**Comment by trdavidson** [comment]
closing this issue due to no response
